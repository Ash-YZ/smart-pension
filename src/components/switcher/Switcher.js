import React from 'react'
import PropTypes from 'prop-types'
import './styles.scss'

const Switcher = props => {
  return (
    <div className='my-3'>
      <h3>Game Mode</h3>
      <span className='switch-label d-none d-md-inline'>{props.gameModes[0]}</span>
      <span className='switch-label d-md-none'>{props.gameModes[0].charAt(0).toUpperCase()}</span>
      <label className="switch">
        <input type="checkbox"
          className='switch-input'
          onChange={props.switchModeHandler} />
        <span className="slider"></span>
      </label>
      <span className='switch-label d-none d-md-inline'>{props.gameModes[1]}</span>
      <span className='switch-label d-md-none'>{props.gameModes[1].charAt(0).toUpperCase()}</span>
    </div>
  )
}

Switcher.propTypes = {
  gameModes: PropTypes.arrayOf(PropTypes.string),
  switchModeHandler: PropTypes.func,
}

export default Switcher
