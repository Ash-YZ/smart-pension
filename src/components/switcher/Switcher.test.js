import React from 'react';
import { shallow } from 'enzyme'
import Switcher from './Switcher'

const changeFn = jest.fn()

const testProps = {
  gameModes: ['people', 'starship'],
  switchModeHandler: changeFn
}

const component = shallow(<Switcher {...testProps} />)

describe('Switcher', () => {
  it('should render game switcher correctly ', () => {
    expect(component).toMatchSnapshot()
  })

  it('should trigger changeFn when the switcher is used', () => {
    component.find('.switch-input').first().simulate('change')
    expect(changeFn).toHaveBeenCalled()
  })
})