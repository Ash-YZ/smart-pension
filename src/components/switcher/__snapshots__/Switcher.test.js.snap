// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switcher should render game switcher correctly  1`] = `
<div
  className="my-3"
>
  <h3>
    Game Mode
  </h3>
  <span
    className="switch-label d-none d-md-inline"
  >
    people
  </span>
  <span
    className="switch-label d-md-none"
  >
    P
  </span>
  <label
    className="switch"
  >
    <input
      className="switch-input"
      onChange={[MockFunction]}
      type="checkbox"
    />
    <span
      className="slider"
    />
  </label>
  <span
    className="switch-label d-none d-md-inline"
  >
    starship
  </span>
  <span
    className="switch-label d-md-none"
  >
    S
  </span>
</div>
`;
